### PR TITLE
adds support for opting out of token syncing when metadata is configured

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- adds support for opting out of token syncing when metadata is configured

## Why are we making these changes?

We wish to allow service owners to explicitly opt-out of token syncing across clusters.

### CODE REVIEW

File: token-syncer/src/token_syncer/commands/syncer.clj [hunk 1]
```diff
@@ -271,28 +275,46 @@
...
-                          (let [sync-result (sync-tokens waiter-api cluster-urls-set limit)
+                          (let [{:keys [details summary] :as sync-result} (sync-tokens waiter-api cluster-urls-set limit metadata-name)
                                 exit-code (-> (get-in sync-result [:summary :sync :failed])
-                                              empty?
-                                              (if 0 1))]
-                            (log/info (-> sync-result pp/pprint with-out-str str/trim))
+                                            empty?
+                                            (if 0 1))]
+                            (log/info "sync details:" (into (sorted-map) details))
+                            (log/info "sync summary:" (into (sorted-map) summary))
```
Comment (shamsimam) [inline token-syncer/src/token_syncer/commands/syncer.clj:303]: Note: Unrelated change to this PR to reduce number if lines output by the syncer.

File: token-syncer/src/token_syncer/commands/syncer.clj [hunk 2]
```diff
@@ -75,7 +75,7 @@
   "Syncs a given token description on all clusters.
    If the cluster-url->token-data says that a given token was not successfully loaded, it is skipped.
    Token sync-ing is also skipped if the tokens are active and the roots are different."
-  [{:keys [store-token]} cluster-urls token latest-token-description cluster-url->token-data]
+  [{:keys [store-token]} cluster-urls token latest-token-description metadata-name cluster-url->token-data]
```
Comment (DaoWen) [inline token-syncer/src/token_syncer/commands/syncer.clj:78]: Would it make sense to rename `metadata-name` to `opt-out-metadata-name`? I think that would make it more immediately clear what that metadata key-value pair is being used for.

Comment (shamsimam) [reply token-syncer/src/token_syncer/commands/syncer.clj]: Renamed to `opt-out-metadata-name`

**Review by DaoWen — APPROVED** [review]
